src/test/java/org/fcrepo/storage/ocfl/DefaultOcflObjectSessionTest.java

```diff
@@ -174,6 +174,36 @@ public void writeUpdateNonRdfResource() {
         assertEquals(2, ocflRepo.describeObject(resourceId).getVersionMap().size());
     }
 
+    @Test
+    public void updateResourceHeaders() {
+        final var resourceId = "info:fedora/foo/bar";
+
+        final var session1 = sessionFactory.newSession(resourceId);
+        final var content1 = atomicBinary(resourceId, "info:fedora/foo", "Test");
+
+        write(session1, content1);
+        session1.commit();
+
+        final var session2 = sessionFactory.newSession(resourceId);
+
+        final var timestamp = Instant.now().plusSeconds(60);
+        final var state = "new-state-token";
+
+        final var existingHeaders = session2.readHeaders(resourceId);
+        final var updatedHeaders = ResourceHeaders.builder(existingHeaders)
+                .withLastModifiedDate(timestamp)
+                .withStateToken(state)
+                .build();
+
+        session2.writeHeaders(updatedHeaders);
+
+        assertEquals(updatedHeaders, session2.readHeaders(resourceId));
```

Comment:
What is intended with the same test before and after the `commit`? I would have almost expected the pre-`commit` test to compare against `existingHeaders`.

Reply:
@awoods When you write to the session, the session will serve the updated content from staging before it is committed to ocfl. The test is written the way it is to confirm that the correct headers are returned when they are retrieved from staging as well as after they are committed to ocfl.

Reply:
Would it be good to have a second session do the same `readHeaders(resourceId)` to verify the existingHeaders are retrieved?

Reply:
Sure

Reply:
Hmmm ok, I would have assumed if the headers have not been committed then anyone outside the session doing the updating would see the old headers. But it appears that once you stage something on an object then that is what everyone sees?

Reply:
When you stage changes in a session those changes are visible within that session but not in any other sessions until they are committed.

Reply:
Oh sorry I missed the line right below these comments. I thought you were doing the staging with `session1` and do the reading of old headers with `session2` but now I see I was reading this wrong.